tensorflow/python/framework/ops.py

```diff
@@ -4737,6 +4737,10 @@ def my_func(pred, tensor):
     """
     if control_inputs is None:
       return self._ControlDependenciesController(self, None)
+
+    # Imported here to avoid circular dependency.
+    from tensorflow.python.ops import resource_variable_ops  # pylint: disable=g-import-not-at-top
```

Comment:
I think this circular import can fail in some internal google things. Let's try though.

Reply:
For the `AutoCastVariable` use case only the func_graph changes are strictly necessary. I only included them here for consistency, but I am happy to revert which would fix any circular dependencies.

Reply:
Let's see what our CI says. If it complains we can revert it.

Reply:
Could we have further explanation of why the `hasattr` check was not sufficient?